controllers/dscinitialization/utils.go

```diff
@@ -18,132 +21,178 @@ import (
-			if err != nil && !k8serr.IsAlreadyExists(err) {
-				log.Error(err, "Unable to create namespace", "name", name)
-				return err
-			}
-		} else {
-			log.Error(err, "Unable to fetch namespace", "name", name)
-			return err
-		}
-		// Patch Application Namespace if it exists
-	} else if dscInit.Spec.Monitoring.ManagementState == operatorv1.Managed {
-		log.Info("Patching application namespace for Managed cluster", "name", name)
-		labelPatch := `{"metadata":{"labels":{"openshift.io/cluster-monitoring":"true","pod-security.kubernetes.io/enforce":"baseline","opendatahub.io/generated-namespace": "true"}}}`
-		err = r.Patch(ctx, foundNamespace, client.RawPatch(types.MergePatchType,
-			[]byte(labelPatch)))
+	// managed cluster need to patch its namespaces
+	if platform == cluster.ManagedRhoai {
+		// Patch default app namespace for Managed cluster
+		err := r.patchAppNS(ctx, nsName, log)
 		if err != nil {
+			log.Error(err, "error patch application namespace for managed cluster")
 			return err
 		}
 	}
-	// Create Monitoring Namespace if it is enabled and not exists
-	if dscInit.Spec.Monitoring.ManagementState == operatorv1.Managed {
-		foundMonitoringNamespace := &corev1.Namespace{}
-		monitoringName := dscInit.Spec.Monitoring.Namespace
-		err := r.Get(ctx, client.ObjectKey{Name: monitoringName}, foundMonitoringNamespace)
-		if err != nil {
-			if k8serr.IsNotFound(err) {
-				log.Info("Not found monitoring namespace", "name", monitoringName)
-				desiredMonitoringNamespace := &corev1.Namespace{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: monitoringName,
-						Labels: map[string]string{
-							labels.ODH.OwnedNamespace: "true",
-							labels.SecurityEnforce:    "baseline",
-							labels.ClusterMonitoring:  "true",
-						},
-					},
-				}
-				err = r.Create(ctx, desiredMonitoringNamespace)
-				if err != nil && !k8serr.IsAlreadyExists(err) {
-					log.Error(err, "Unable to create namespace", "name", monitoringName)
-					return err
-				}
-			} else {
-				log.Error(err, "Unable to fetch monitoring namespace", "name", monitoringName)
-				return err
-			}
-		} else { // force to patch monitoring namespace with label for cluster-monitoring
-			log.Info("Patching monitoring namespace", "name", monitoringName)
-			labelPatch := `{"metadata":{"labels":{"openshift.io/cluster-monitoring":"true", "pod-security.kubernetes.io/enforce":"baseline","opendatahub.io/generated-namespace": "true"}}}`
-
-			err = r.Patch(ctx, foundMonitoringNamespace, client.RawPatch(types.MergePatchType, []byte(labelPatch)))
-			if err != nil {
-				return err
-			}
-		}
+	// Patch monitoring namespace
+	err := r.patchMonitoringNS(ctx, dscInit, log)
+	if err != nil {
+		log.Error(err, "error patch monitoring namespace for managed cluster")
+		return err
 	}
 
 	// Create default NetworkPolicy for the namespace
-	err = r.reconcileDefaultNetworkPolicy(ctx, name, dscInit, platform)
+	err = r.reconcileDefaultNetworkPolicy(ctx, nsName, dscInit, platform, log)
```

Comment:
the logger should be part of the context, it should not be required to explicitly pass it to the functions